addons/pos_online_payment_self_order/models/pos_config.py

```diff
@@ -22,3 +22,9 @@ def _get_self_ordering_data(self):
         payment_methods = self.self_order_online_payment_method_id.read(payment_search_params['search_params']['fields'])
         res['pos_payment_methods'] += payment_methods
         return res
+
+    def _get_allowed_payment_methods(self):
```

Comment:
it looks very close to the one in pos_self_order. 
Can you do it in a way you call super?